### PR TITLE
EVE Discovery service: move `allow_to_discover` to AppInst port 

### PR DESCRIPTION
**Pull request by uncleDecart**
This is more granular approach. From appInstance perspective,
you also know which network port will be used.

### CODE REVIEW

**Comment by uncleDecart** [comment]
I see yetus [error](https://github.com/lf-edge/eve-api/actions/runs/8701137540/job/23862550960?pr=52#step:4:260) 

```
bufcompat:Previously present field "24" with name "allow_to_discover" on message "AppInstanceConfig" was deleted.
```

But I marked field 24 as reserved, not sure we should leave it there

File: proto/config/appconfig.proto [hunk 1]
```diff
@@ -103,6 +103,9 @@ message SnapshotConfig {
 // can also be increased to cause an application instance restart without
 // any other change to the application instance.
 message AppInstanceConfig {
+  // deprecated fields
+  reserved 14, 24;
```
Comment (eriknordmark) [inline proto/config/appconfig.proto:107]: How about also removing the comment with   // Deprecated 14

File: proto/config/netconfig.proto [hunk 2]
```diff
@@ -55,6 +55,10 @@ message NetworkAdapter {
   // valid vlan id range: 2 - 4093
   // vlan id 1 is implicitly used by linux bridges
   uint32 access_vlan_id = 41;
+
+  // allow AppInstance to discover other AppInstances
+  // attached to its network instances. default is false
```
Comment (eriknordmark) [inline proto/config/netconfig.proto:60]: Shouldn't this say "to this network instance" in singular?
Or to the network instance attached to this adapter?

**Review by eriknordmark — APPROVED** [review]
LGTM but some comment nits

File: proto/config/netconfig.proto (hunk 2 above)
Comment (uncleDecart) [reply proto/config/netconfig.proto]: renamed to "this network instance"

**Review by milan-zededa — APPROVED** [review]

**Pull request merged**